yellow-paper/docs/cryptography/protocol-overview.md

```diff
@@ -0,0 +1,113 @@
+# Proving System Components
+
+# Interactive Proving Systems
+
+## Ultra Plonk
+
+UltraPlonk is a variant of the [PLONK](https://eprint.iacr.org/2019/953) protocol - a zkSNARK with a universal trusted setup.
+
+UltraPlonk utilizes the "Ultra" circuit arithmetisation. This is a configuration with four wires per-gate, and the following set of gate types:
+
+- arithmetic gate
+- elliptic curve point addition/doubling gate
+- range-check gate
+- plookup table gate
+- memory-checking gates
+- non-native field arithmetic gates
+
+## Honk
+
+Honk is a variant of the PLONK protocol. Plonk performs polynomial testing via evaluating a polynomial relation is zero modulo the vanishing polynomial of a multiplicative subgroup. Honk performs the polynomial testing via evaluating, using a sumcheck protocol, that a relation over multilinear polynomials vanishes when summed over a boolean hypercube.
+
+The first protocol to combine Plonk and the sumcheck protocol was [HyperPlonk](https://eprint.iacr.org/2022/1355)
+
+Honk uses a custom arithmetisation that extends the Ultra circuit arithmetisation (not yet finalized)
+
+# Incrementally Verifiable Computation Subprotocols
+
+An Incrementally Verifiable Computation (IVC) scheme describes a protocol that enables multiple successive proofs to evolve the value taken by some defined persistent state over time.
+
+IVC schemes are used by Aztec in two capacities:
+
+1. to compute a client-side proof of one transaction execution.
+2. to compute a proof of a "rollup" circuit, that updates rollup state based on a block of user transactions
+
+Both use IVC schemes. Client-side, each function call in a transaction is a "step" in the IVC scheme. Rollup-side, aggregating two transaction proofs is a "step" in the IVC scheme.
+
+The client-side IVC scheme is substantially more complex than the rollup-side scheme due to performance requiremenmts.
+
+Rollup-side, each "step" in the IVC scheme is a Honk proof, which are recursively verified. As a result, no protoocols other than Honk are required to execute rollup-side IVC.
+
+We perform one layer of "proof-system compression" in the rollup. The final proof of block-correctness is constructed as a Honk proof. An UltraPlonk circuit is used to verify the correctness of the Honk proof, so that the proof that is verified on-chain is an UltraPlonk proof (verification gas costs are lower for UltraPlonk vs Honk).
```

Comment:
(verification gas costs are lower for UltraPlonk vs Honk) - I would state briefly why this is the case

Reply:
done